src/services/codeFixProvider.ts

```diff
@@ -18,22 +26,38 @@ namespace ts {
 
     export namespace codefix {
         const codeFixes: CodeFix[][] = [];
+        const diagnosticGeneratingCodeFixes: CodeFix[] = [];
 
-        export function registerCodeFix(action: CodeFix) {
-            forEach(action.errorCodes, error => {
+        export function registerCodeFix(codeFix: CodeFix) {
+            forEach(codeFix.errorCodes, error => {
                 let fixes = codeFixes[error];
                 if (!fixes) {
                     fixes = [];
                     codeFixes[error] = fixes;
                 }
-                fixes.push(action);
+                fixes.push(codeFix);
             });
+
+            if (codeFix.createCodeFixDiagnosticIfApplicable) {
+                diagnosticGeneratingCodeFixes.push(codeFix);
+            }
         }
 
         export function getSupportedErrorCodes() {
             return Object.keys(codeFixes);
         }
 
+        export function getCodeFixDiagnosticsForNode(context: CodeFixDiagnoseContext, node: Node): Diagnostic[] | undefined {
+            let result: Diagnostic[];
+            for (const codeFix of diagnosticGeneratingCodeFixes) {
+                const newDiag = codeFix.createCodeFixDiagnosticIfApplicable(node, context);

```

Comment:
This member is declared as optional, but isn't checked for `undefined` in this method